pkg/gadgets/run/tracer/run.go

```diff
@@ -52,8 +52,9 @@ func (g *GadgetDesc) Category() string {
 }
 
 func (g *GadgetDesc) Type() gadgets.GadgetType {
-	// Currently trace only
-	return gadgets.TypeTrace
+	// Placeholder for gadget type. The actual type is determined at runtime by using
+	// GetGadgetInfo()
+	return gadgets.TypeRun
```

Comment:
You're absolutely right. The main issue is that we need to support built-in and containerized gadgets at the same time, so we're trying to use as much as possible existing infrastructure to avoid having to implement all of that from scratch. This creates this kind of odd situations where a concept doesn't fit very well with containerized gadgets. 

I already discussed with @flyth and we agreed to keep trying to use the existing code, later on, when we'll be remove built-in gadgets, we'll be able to perform a big cleanup.